crates/bevy_ecs/src/archetype.rs

```diff
@@ -342,46 +481,60 @@ impl Archetype {
             .map(|info| info.archetype_component_id)
     }
 
+    /// Clears all entities from the archetype.
     pub(crate) fn clear_entities(&mut self) {
         self.entities.clear();
     }
 }
 
-/// A generational id that changes every time the set of archetypes changes
-#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+/// An opaque generational id that changes every time the set of [`Archetypes`] changes.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
 pub struct ArchetypeGeneration(usize);
 
 impl ArchetypeGeneration {
     #[inline]
-    pub const fn initial() -> Self {
+    pub(crate) const fn initial() -> Self {
         ArchetypeGeneration(0)
     }
 
     #[inline]
-    pub fn value(self) -> usize {
+    pub(crate) fn value(self) -> usize {
         self.0
     }
 }
 
 #[derive(Hash, PartialEq, Eq)]
-pub struct ArchetypeIdentity {
+struct ArchetypeIdentity {
     table_components: Box<[ComponentId]>,
     sparse_set_components: Box<[ComponentId]>,
 }
 
+/// An opaque unique joint ID for a [`Component`] in an [`Archetype`] within a [`World`].
+///
+/// A component may be present within multiple archetypes, but each of those components within
+/// each archetype has its own unique `ArchetypeComponentId`. This is leveraged by the system
+/// schedulers to opprotunistically run multiple systems in parallel that would otherwise
+/// conflict. For example, `Query<&mut A, With<B>>` and `Query<&mut A, Without<B>>` can run in
+/// parallel as the matched `ArchetypeComponentId` sets for both queries are disjoint, even
+/// though `&mut A` on both queries point to the same [`ComponentId`].
+///
+/// Every [`Resource`] is also assigned one.
```

Comment:
Nit: One what? I assume you mean an ArchetypeComponentId, but resources are not components right?

Reply:
Resources are also assigned ComponentIds, so in a way it does. Expanded on this a bit more.